src/domain/child/ChildEnrolmentCount.tsx

```diff
@@ -0,0 +1,52 @@
+import { useTranslation } from 'react-i18next';
+import { IconCheck, LoadingSpinner } from 'hds-react';
+import { useQuery } from '@apollo/client';
+
+import {
+  ChildEnrolmentCount as ChildEnrolmentCountQuery,
+  ChildEnrolmentCountVariables as ChildEnrolmentCountVariablesQuery,
+} from '../api/generatedTypes/childEnrolmentCount';
+import KukkuuPill from '../../common/components/kukkuuPill/KukkuuPill';
+import { childEnrolmentCountQuery } from './queries/ChildEnrolmentCountQuery';
+
+type Props = {
+  childId: string;
+};
+
+export default function ChildEnrolmentCount({ childId }: Props) {
+  const { t } = useTranslation();
+  const { data } = useQuery<
+    ChildEnrolmentCountQuery,
+    ChildEnrolmentCountVariablesQuery
+  >(childEnrolmentCountQuery, {
+    variables: {
+      childId,
+    },
+  });
+
+  const pastEnrolmentCount = data?.child?.pastEnrolmentCount ?? ' ';
+  const enrolmentLimit = data?.child?.project?.enrolmentLimit ?? ' ';
+  const areAllEnrolmentsUsed = data && pastEnrolmentCount === enrolmentLimit;
+
+  return (
+    <KukkuuPill
+      variant={areAllEnrolmentsUsed ? 'success' : 'default'}
+      iconLeft={areAllEnrolmentsUsed && <IconCheck />}
+      name={
+        <>
+          {t('child.message.eventVisitsThisYear')}{' '}
+          {data ? (
+            `${pastEnrolmentCount}/${enrolmentLimit}`
+          ) : (
+            <LoadingSpinner
+              theme={{
+                '--spinner-color': 'var(--color-black)',
+              }}
+              small
+            />
+          )}
+        </>
+      }
+    />
```

Comment:
Quite hard to read... 🤔

Reply:
could the loading spinner and the actual text content be in variables? Then they could be assigned to the text with a string-formatter `${t('child.message.eventVisitsThisYear')} ${data ? counter ? loadingspinner}` and there would be no need for the empty fragment either.

Reply:
The spinners started to work better later on. Some problems with infra. The pill component could look less like a button when there would be a colon: "Tapahtumakäynnit tänä vuonna: ..."

Reply:
I don't mind making this change. I'd perhaps personally prefer your version, but it often gets flagged as unclear as well. Some devs find it difficult to understand because you have to "jump around" on the file and you can't just browse it from top to bottom.

I added the colon.

Reply:
Unfortunately we can't use string templates/concatenation here as the spinner's as the "primitive" type a component maps to is a function that can't be stringified by default. I implemented your suggestion to the degree it was possible.